jetson/nav/search/searchStateMachine.cpp

```diff
@@ -284,28 +285,92 @@ NavState SearchStateMachine::executeSearchDrive( Rover* phoebe, const rapidjson:
 } // executeSearchDrive()
 
 // Executes the logic for turning to the target.
+// Starts by turnign the gimbal towards the target. Then holds for 200 cycles to confirm aquisition
 // If the rover loses the target, will continue to turn using last known angles.
+// Once target location confirmed, rover turns towards the target, then rights its gimbal
 // If the rover finishes turning to the target, it goes into waiting state to
 // give CV time to relocate the target
 // Else the rover continues to turn to to the target.
-NavState SearchStateMachine::executeTurnToTarget( Rover* phoebe )
+NavState SearchStateMachine::executeTurnToTarget( Rover* phoebe, const rapidjson::Document& roverConfig  )
 {
-    if( phoebe->roverStatus().leftTarget().distance < 0 )
+    static int turnToTargetStage = 0;
+    static double bearingAngle = 0;
+
+    static bool firstTime = true;
+    
+    //if we are not using a gimbal, just skip all gimbal states and immeaditly turn to target, only needs to run once
+    if ( !roverConfig[ "search" ][ "useGimbal" ].GetBool() && firstTime){
+        turnToTargetStage = 2;
+        firstTime = false;
+        bearingAngle = phoebe->roverStatus().leftTarget().bearing + phoebe->roverStatus().odometry().bearing_deg;
+    }
+    
+    if( phoebe->roverStatus().leftTarget().distance < 0  && turnToTargetStage != 2 && turnToTargetStage != 3)
     {
         cerr << "Lost the target. Continuing to turn to last known angle\n";
         if( phoebe->turn( mTargetAngle + mTurnToTargetRoverAngle ) )
         {
+            turnToTargetStage = 0;
+            bearingAngle = 0;
+            firstTime = true;
             return NavState::TurnedToTargetWait;
         }
         return NavState::TurnToTarget;
     }
-    if( phoebe->turn( phoebe->roverStatus().leftTarget().bearing +
-                      phoebe->roverStatus().odometry().bearing_deg ) )
-    {
+   
+    if ( turnToTargetStage == 0 ){
```

Comment:
can you comment what each stage does? just one line for each is fine. I see it's kinda outlined in the function description, but specificity in the implementation here would help